### ISSUE TITLE
Individual Permission on Data Package Entities

### ISSUE THREAD

**Issue by p-a-s-c-a-l** [clarity-h2020/docker-drupal#82]
Some Data Packages and the related Resources , e.g. the Expert Data Packages for the DCs or other "premium" Data Packages shall be available for users of the role "subscribers" only. Since we cannot assign individual permissions on single entity / node level, we have to use modules like [Permissions by Term](https://www.drupal.org/project/permissions_by_term).

ATM, this issue has low priority and becomes relevant when our first "premium" or DC-level Data Package is available. 

Furthermore, [Single-Sign-On](https://github.com/clarity-h2020/docker-drupal/issues/16) has to be supported and [Expert Profiles](https://github.com/clarity-h2020/marketplace/issues/2) have to be implemented @fgeyer16.

**Comment by DenoBeno** [comment]
I also believe that this isn't a priority. If and when we decide to implement this, it can probably be done using the https://www.drupal.org/project/permissions_by_term module.

**Comment by patrickkaleta** [comment]
Since I now already installed the **permissions by term** module, I tested the individual permissions locally and suggest the following approach:

**1)** create new User roles with (for the start) rather generic terms like: `Level - DC` or `Level - 1`, ...
**2)** create new taxonomy "Permission levels" with terms like in **(1)** and assign the appropriate user roles
**3)** create a new field ("Permissions" or "Access permissions",...) in the data package and reference the new taxonomy
**4)** when creating a new data package the editor selects the permissions needed (or leave this field empty, so that it can be used by all users)

Then, in the "Study data" tab users can only see and select the data packages corresponding to their user roles.

**I now see the following open questions/problems:**
- I'm not sure what will happen when we assign permissions to individual  resources as well and how our system will then react if the permission of a resource doesn't match the permission of its data package.
- what about member of a study with different permission levels? One member could select a data package, that would be not accessible by others. To them it would look like the study still doesn't have a data package selected. IMO the Study owner needs to make sure that his/her team all have the same permission level (could be difficult to manage).
